src/Avalonia.Base/Threading/DispatcherOperation.cs

```diff
@@ -311,7 +306,19 @@ protected virtual void InvokeCore()
 
     internal virtual object? GetResult() => null;
     
-    protected virtual void AbortTask() => (TaskSource as TaskCompletionSource<object?>)?.SetCanceled();
+    protected virtual void AbortTask()
+    {
+        object? taskSource;
+        lock (Dispatcher.InstanceLock)
```

Comment:
This logic needs to be applied to the overriden `DispatcherOperation<T>.AbortTask()`. Consider making the method non-virtual and pass the captured `taskSource` to a virtual method responsible for calling `SetCanceled`.

Reply:
Overridden method always has a valid task source since it's created in ctor.

Reply:
https://github.com/AvaloniaUI/Avalonia/blob/3d2dc202383362d603eae0e1ceb538f3e0d87591/src/Avalonia.Base/Threading/DispatcherOperation.cs#L350-L354